text/0026-log-levels.md

```diff
@@ -0,0 +1,57 @@
+# Common Logging Levels for Buildpacks
+
+## Summary
+
+All Paketo buildpacks should provide a mechanism for tuning log verbosity. To
+enable this configuration, the buildpacks should respect a `BP_LOG_LEVEL`
+environment variable.
+
+## Motivation
+
+The Paketo buildpacks aim to provide transparent and understandable feedback to
+buildpack users. Part of this feedback is provided by the log output emitted by
+each buildpack. Buildpack users and authors will want see log output with
+different levels of granularity in the details they provide. For instance, some
+buildpack users may only want to know when the buildpack is warning the user
+about possible issues. Alternatively, to aid in some debugging scenarios, it
+may be useful to emit log output that including internal implementation details
+and state. Clearly providing log output that succeeds in serving both of these
+needs is not possible. Instead, we should allow buildpack users to tune the log
+output to suite their needs.
+
+## Detailed Explanation
+
+All buildpacks will provide a `BP_LOG_LEVEL` environment variable to allow for
+the configuration of log output. This variable will support 3 acceptable
+options: `WARN`, `INFO`, and `DEBUG`. The default option, should none be
+provided by the user, will be `INFO`. While this RFC will not outline detailed
+specifics for what logs should appear at each level, a basic description of
```

Comment:
I agree that the specifics of what should be logged shouldn't be covered here, but maybe there should be some standardization across what the formats should look like at some point. It's a bit early for this, but it would be nice to have this laid out somehow so that if it is implemented in both `packit` and `libpak` for example, we have a paper trail of what that agreed format should be. 

The logging across some of the Java buildpacks is already different from the logging in other buildpacks in terms of color output (for example, see the CA Certificate Buildpack logs when it runs in the Node language family) it would be nice if we could minimize any other differences ahead of time.

Reply:
This RFC is meant to be a start for negotiating what a common log output might look like for all of the buildpacks. I want this RFC and others that follow on this path to be small so that we aren't debating some massive omnibus RFC that overhauls the entirety of log output. I think we all agree that we would like to get to a common log output, but it will be quite a bit of work. I'd like to see us break the problem down into reasonable pieces so that we can iterate towards a solution.